src/views/design/ScriptDetail/EditAction/index.tsx

```diff
@@ -226,26 +240,44 @@ function EditAction({ onClose, action, onEdit, state }: IPublicProps & IObserveP
                                 setErrorExpectedChecked(!errorExpectedChecked);
                             }}
+                            disabled={!checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')}
                         />
                     </Box>
 
                     <Box marginLeft={2}>
                         <ButtonGroup size="small">
-                            <Button
-                                color="default"
-                                variant="outlined"
-                                onClick={onClose}
-                            >
-                                <Translate msg="scripts.detail.edit_action.footer.cancel" />
-                            </Button>
-                            <Button
-                                variant="contained"
-                                color="secondary"
-                                disableElevation
-                                onClick={updateAction}
-                            >
-                                <Translate msg="scripts.detail.edit_action.footer.save" />
-                            </Button>
+                            {!checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')
+                            && checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_READ, 'PUBLIC')
+                                ? (
+                                    <Button
+                                        color="default"
+                                        variant="outlined"
+                                        onClick={onClose}
+                                    >
+                                        <Translate msg="scripts.detail.edit_action.footer.go_back" />
+                                    </Button>
+                                )
+                                : (
+                                    <Button
+                                        color="default"
+                                        variant="outlined"
+                                        onClick={onClose}
+                                    >
+                                        <Translate msg="scripts.detail.edit_action.footer.cancel" />
+                                    </Button>
+                                )}
+                            {checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')
+                                ? (
+                                    <Button
+                                        variant="contained"
+                                        color="secondary"
+                                        disableElevation
+                                        onClick={updateAction}
+                                    >
+                                        <Translate msg="scripts.detail.edit_action.footer.save" />
+                                    </Button>
+                                )
+                                : null}
```

Comment:
Can we already use the security group name  of the script in question instead of the fixed value 'PUBLIC'.
Also, refactor the logic
```
if (checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_WRITE, 'PUBLIC')) {
    logic
} else {
    if (checkAuthority(SECURITY_PRIVILEGES.S_SCRIPTS_READ, 'PUBLIC')) {
         logic
    } else {
         null
    }
}
```